### ISSUE TITLE
Disabling focus manager takes two clicks

### ISSUE THREAD

**Issue by dlech** [palantir/blueprint#5229]
#### Environment

- __Package version(s)__: @blueprintjs/core 4.0.4
- __Operating System__: Ubuntu 20.04
- __Browser name and version__: Firefox 98, Chromium 100

#### Code Sandbox

https://codesandbox.io/s/blueprint-sandbox-forked-4x77pn?file=/src/index.tsx

#### Steps to reproduce

1. Click on the *Always focus* button 
   - note the focus outline on this button - this is expected 
   - note the state text - this is expected
1. Click on the *Focus on tabs only* button -
   - note the focus outline on this button - this is _not_ expected
   - note the state text - this _is_ expected and tell us that `FocusStyleManager.onlyShowFocusOnTabs()` was called
1. Click on the *Focus on tabs only* button 
   - note the focus outline is gone - this should have happened on the previous click, not this one

#### Actual behavior

The style change is not applied until the click after `FocusStyleManager.onlyShowFocusOnTabs()` is called.

#### Expected behavior

The style change should be applied immediately when `FocusStyleManager.onlyShowFocusOnTabs()` is called.

#### Possible solution

I think a `mousedown` event is already queued when `FocusStyleManager.onlyShowFocusOnTabs()` is called so it immediately fires and applies the style change again. We could add an `if (this.isRunning)` in the `mousedown` handler to make sure it doesn't do this if the style manager has been disabled in the meantime.

**Comment by adidahiya** [comment]
Can you explain your use case a bit more? Why do you have to switch between the different focus manager modes interactively? Why do you expect this to change during the lifetime of your application or web page?

**Comment by dlech** [comment]
I am using https://github.com/lukasbach/react-complex-tree to get a keyboard accessible tree component since the Blueprints tree is not keyboard accessible. When the tree widget has focus, we want the focus manager to always show focus so we can see which item it the tree is focused but when the tree widget is not enabled we don't want to show the focus outline on other components.

**Comment by dlech** [comment]
> I think a `mousedown` event is already queued when `FocusStyleManager.onlyShowFocusOnTabs()` is called so it immediately fires and applies the style change again. We could add an `if (this.isRunning)` in the `mousedown` handler to make sure it doesn't do this if the style manager has been disabled in the meantime.

This guess turns out to be wrong. The problem happens when `InteractionModeEngine.start()` is called rather than `stop()`. This means that we have to know what state we should be in when `start()` is called. So I suppose we could add an optional parameter to `FocusStyleManager.onlyShowFocusOnTabs()` and pass it through to `InteractionModeEngine.start()`.

**Comment by adidahiya** [comment]
> When the tree widget has focus, we want the focus manager to always show focus so we can see which item it the tree is focused but when the tree widget is not enabled we don't want to show the focus outline on other components.

Another solution might be to add an escape hatch -- some new CSS class which Blueprint's focus style manager knows not to mess with, so that it can continue to show focus outlines even after `.onlyShowFocusOnTabs()` is enabled...

**Comment by dlech** [comment]
Yes, that sounds like a good idea, then we wouldn't have to change `FocusStyleManager` at runtime. My second suggestion doesn't really work since you would have to know if the new element receiving focus is a regular element or one of the exempt elements.
